includes/Core/Storage/Setting_With_Owned_Keys_Trait.php

```diff
@@ -51,21 +49,53 @@ function ( $option, $value ) use ( $keys ) {
 
 		add_filter(
 			'pre_update_option_' . static::OPTION,
-			function( $value, $old_value ) use ( $keys ) {
+			function ( $value, $old_value ) {
 				if ( is_array( $value ) && is_array( $old_value ) ) {
-					foreach ( $keys as $key ) {
-						if ( isset( $value[ $key ], $old_value[ $key ] ) && $value[ $key ] !== $old_value[ $key ] && current_user_can( Permissions::MANAGE_OPTIONS ) ) {
-							$value['ownerID'] = get_current_user_id();
-							break;
-						}
-					}
+					return $this->maybe_update_owner_id_in_settings_before_saving_them( $value, $old_value );
 				}
-
-				return $value;
 			},
 			10,
 			2
 		);
 	}
 
+	/**
+	 * Updates settings to set the current user as an owner of the module if settings contain owned keys.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $settings The module settings.
+	 */
+	protected function maybe_set_owner_id( $settings ) {
+		$keys = $this->get_owned_keys();
+		if ( count( array_intersect( array_keys( $settings ), $keys ) ) > 0 && current_user_can( Permissions::MANAGE_OPTIONS ) ) {
+			$this->merge( array( 'ownerID' => get_current_user_id() ) );
+		}
+	}
+
+	/**
+	 * Updates the current module settings to have the current user as the owner of the module if at least
+	 * one of the owned keys have been changed.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $settings The new module settings.
+	 * @param array $old_settings The old module settings.
+	 */
+	protected function maybe_update_owner_id_in_settings_before_saving_them( $settings, $old_settings ) {
```

Comment:
Similarly, this name could be a bit more clear because it doesn't actually update the value in the DB but "calculates" or filters the given value.

How about something like `filter_owner_id_for_updated_settings`?

Reply:
Renamed it.